### PR TITLE
download: utils to download files and blobs

### PR DESCRIPTION
**Pull request by ahmetalpbalkan**
This one is fairly simple:

```

package download

FUNCTIONS

func Download(d Downloader) (io.ReadCloser, error)
    Download retrieves a response body and checks the response status code
    to see if it is 200 OK and then returns the response body. It issues a
    new request every time called. It is caller's responsibility to close
    the response body.

TYPES

type Downloader interface {
    // GetRequest returns a new GET request for the resource.
    GetRequest() (*http.Request, error)
}
    Downloader describes a method to download files.

func NewBlobDownload(accountName, accountKey string, blob blobutil.AzureBlobRef) Downloader
    NewBlobDownload creates a new Downloader for a blob hosted in Azure Blob
    Storage.

func NewURLDownload(url string) Downloader
    NewURLDownload creates a new downloader with the provided URL

```

### CODE REVIEW

File: download/blob_test.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+import (
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/storage"
+	"github.com/ahmetalpbalkan/custom-script-extension-linux/blobutil"
```
Comment (boumenot) [inline download/blob_test.go]: ahmetalpbalkan -> Azure?

**Comment by boumenot** [comment]
LGTM

File: download/blob_test.go (hunk 1 above)
Comment (ahmetalpbalkan) [inline download/blob_test.go]: I'll update these altogether.

**Comment by ahmetalpbalkan** [comment]
@boumenot actually one thing to note is, in this rewrite I switched from authenticating to blob storage and getting the blob (via the Authorization header crafted by the storage client) to downloading the blob from a SAS URL (crafted by the storage client) that is generated freshly every time and is good 30 minutes. I do not foresee any issues with this, let me know if you can think of any. They should work the same.

**Pull request merged**